packages/extract-text/index.js

```diff
@@ -19,26 +19,28 @@ function extractText (outputFilePattern, fileType) {
 
   const plugin = new ExtractTextPlugin(outputFilePattern)
 
-  return (context, webpackConfig) => {
-    const loaderConfig = getLoaderConfigByType(context, webpackConfig, fileType)
+  return (context, util) => prevConfig => {
+    const loaderConfig = getLoaderConfigByType(context, prevConfig, fileType)
     const nonStyleLoaders = getNonStyleLoaders(loaderConfig, fileType)
 
-    return {
-      module: {
-        loaders: [
-          {
-            test: context.fileType(fileType),
-            exclude: loaderConfig.exclude,
-            loader: plugin.extract({
-              fallbackLoader: 'style-loader',
-              loader: nonStyleLoaders
-            }),
-            loaders: undefined
-          }
-        ]
-      },
-      plugins: [ plugin ]
-    }
+    // Partial application of `addLoader`, `addPlugin` & `removeLoaders`
+    // Bind them to their job, but don't apply them on a config yet
+
+    const _addLoader = util.addLoader({
+      test: context.fileType(fileType),
+      exclude: loaderConfig.exclude,
+      loader: plugin.extract({
+        fallbackLoader: 'style-loader',
+        loader: nonStyleLoaders
+      })
+    })
+    const _addPlugin = util.addPlugin(plugin)
+    const _removeLoaders = removeLoaders(context.fileType(fileType))
+
+    // Now apply them to the config: Remove the existing loaders for that
+    // file type, add the new loader and add the plugin
+
+    return _addPlugin(_addLoader(_removeLoaders(prevConfig)))
```

Comment:
But it isn't a group of blocks, it's just three update functions... I already thought about `util.aggregate([ _removeLoaders, ... ])` or something like that, but I'm hesitating adding more new stuff to `util`.